### ISSUE TITLE
Bug: On status changed - Unhandled Exception: Bad state: Future already completed

### ISSUE THREAD

**Issue by gabdsg** [ShivamArora/meteorify#2]
```
_client.addStatusListener((status) {
      if (status == ConnectStatus.connected) {
        isConnected = true;
        _notifyConnected();
        completer.complete(ConnectionStatus.CONNECTED);
      } else if (status == ConnectStatus.disconnected) {
        isConnected = false;
        _notifyDisconnected();
        completer.completeError(ConnectionStatus.DISCONNECTED);
      }
    });
```

When the meteor status changes, the StatusListener emits a new value and the future calls the completer again.

**Comment by ShivamArora** [comment]
Thanks for reporting this up!

I'll look into this issue and fix this soon.

**Comment by ShivamArora** [comment]
@gabdsg This bug has been fixed in the recent release `^1.0.3`
